src/main/java/com/velox/sloan/cmo/workflows/kapalibrary/AutoIndexAssignmentHelper.java

```diff
@@ -0,0 +1,141 @@
+package com.velox.sloan.cmo.workflows.kapalibrary;
+
+import com.velox.api.datarecord.NotFound;
+import com.velox.sapioutils.shared.managers.ManagerBase;
+
+import java.rmi.RemoteException;
+
+
+/**
+ * This is a helper class with common methods used across plugins. Index Barcode and Adapter terms are used interchangeably and have the same meaning.
+ * 'AutoIndexAssignmentConfig' is the DataType which holds the Index Barcode metadata that is used for Auto Index Assignment to the samples.
+ *
+ * @author sharmaa1
+ */
+
+public class AutoIndexAssignmentHelper extends ManagerBase {
+
+    /**
+     * Get the minimum liquid volume for the Index Barcode to be present in a plate well for a plate given its plate size.
+     *
+     * @param plateSize
+     * @return Double volume
+     */
+    public Double getMinAdapterVolumeRequired(Integer plateSize) {
+        if (plateSize == 96) {
+            return 7.50;
+        } else {
+            return 3.00;
+        }
+    }
+
+    /**
+     * Get the volume of Index Barcode to transfer to a well on new plate.
+     *
+     * @param startingAdapterConcentration
+     * @param minAdapterVolume
+     * @param targetAdapterConcentration
+     * @return Double volume
+     */
+    public Double getAdapterInputVolume(Double startingAdapterConcentration, Double minAdapterVolume, Double targetAdapterConcentration) {
+        Double c1 = startingAdapterConcentration;
+        Double v1 = minAdapterVolume;
+        Double c2 = targetAdapterConcentration;
+        Double v2 = (c2 * v1) / c1;
+        if (v2 >= 2.00) {
+            return v2;
+        }
+        return 2.00;
+    }
+
+
+    /**
+     * Get concentration of adapter in the new plate given the DNA Input amount.
+     *
+     * @param dnaInputAmount
+     * @return Double Concentration
+     */
+    public Double getCalculatedTargetAdapterConcentration(Double dnaInputAmount, Integer plateSize) {
+        if (plateSize == 96 && dnaInputAmount < 50.00) {
+            return (15 * dnaInputAmount) / 50.00;
+        }
+        if (plateSize == 384 && dnaInputAmount < 20.00) {
+            return (15 * dnaInputAmount) / 20.00;
+        }
+        return 15.00;
+    }
+
+    /**
+     * Get the volume of water to transfer to new plate.
+     *
+     * @param startingAdapterConcentration
+     * @param minAdapterVolume
+     * @param targetAdapterConcentration
+     * @param maxPlateVolume
+     * @return Double volume
+     */
+    public Double getVolumeOfWater(Double startingAdapterConcentration, Double minAdapterVolume, Double targetAdapterConcentration, Double maxPlateVolume) {
```

Comment:
add unit tests, possibly "public static" so no instance of the class is necessary to test.